### DISCUSSION TITLE
Shape of auth.users.raw_user_meta_data across providers

### DISCUSSION THREAD

**Discussion by malerba118** [supabase/supabase#5210] [General]
So the bottom line is I'm wondering if the shape of auth.users.raw_user_meta_data will be consistent across social auth providers or if it will vary depending on the provider?

What i need to do right now is basically create a default user view from the auth.users table that includes an avatar and a display name.

So far here's what i'm trying and it's working fairly well

```sql
create or replace view display_users as select 
  id, 
  coalesce(raw_user_meta_data ->> 'name', raw_user_meta_data ->> 'full_name', raw_user_meta_data ->> 'user_name') as name, 
  coalesce(raw_user_meta_data ->> 'avatar_url', raw_user_meta_data ->> 'avatar') as avatar 
from auth.users;
```

but i'm wondering what other fields might exist on raw_user_meta_data that could include valuable avatar/name information (eg `picture`, `profile_image`, `username`, `display_name`)

I don't want to go through the process of actually setting up every social auth provider so if anyone has examples of what the raw_user_meta_data object looks like for various providers i would very much appreciate if you could share those with me just so i can get a sense of the different shapes i might encounter for this data structure.

Thanks!

**Comment by kevinmlong** [comment]
I just ran into this earlier today and was wondering the same thing. In my search for learning more, I came across this discussion. Would be very interested in what the Supabase team could provide in terms of documentation for this field.
